### PR TITLE
DAOS-9989 test: Enable weekly testing with tcp and ucx providers

### PR DESCRIPTION
**Pull request by phender**
Enable weekly testing of the pr and daily_regession taggged tests on the
Functional HW test stages with the tcp and ucx providers.  This extends
the normal CI daily testing of pr and daily_regession tagged tests on
the Functional HW test stages with the verbs provider.

Signed-off-by: Phillip Henderson <phillip.henderson@intel.com>

### CODE REVIEW

**Comment by brianjmurrell** [comment]
I think you might have/want to change `$BRANCH_NAME` here https://github.com/daos-stack/daos/blob/9c4ac298bf33dfdd4f091ccc2b309796141112cd/ci/provisioning/post_provision_config_common_functions.sh#L196 to `$CHANGE_BRANCH`.

File: Jenkinsfile [hunk 1]
```diff
@@ -14,6 +14,7 @@
 // To use a test branch (i.e. PR) until it lands to master
 // I.e. for testing library changes
 //@Library(value="pipeline-lib@your_branch") _
+@Library(value="pipeline-lib@DAOS-9989") _
```
Comment (brianjmurrell) [inline Jenkinsfile:17]: Just noting that this cannot land until this is backed out.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
[As noted](https://github.com/daos-stack/daos/pull/8902#issuecomment-1120208814) you also need to make a change to https://github.com/daos-stack/daos/blob/9c4ac298bf33dfdd4f091ccc2b309796141112cd/ci/provisioning/post_provision_config_common_functions.sh#L196 to replace `BRANCH_NAME` with `CHANGE_BRANCH` to get the post-provisioning to actually work.

File: Jenkinsfile (hunk 1 above)
Comment (phender) [reply Jenkinsfile:17]: Set to draft until pipeline-lib change is landed

File: Jenkinsfile [hunk 2]
```diff
@@ -80,9 +80,6 @@ pipeline {
         string(name: 'BaseBranch',
                defaultValue: base_branch,
                description: 'The base branch to run weekly-testing against (i.e. master, or a PR\'s branch)')
-        string(name: 'CI_RPM_TEST_VERSION',
-               defaultValue: '',
-               description: 'Package version to use instead of building. example: 1.3.103-1, 1.2-2')
```
Comment (brianjmurrell) [inline Jenkinsfile]: Why is this being removed.  I use this.

**Comment by phender** [comment]
I need to investigate why the Functional HW stages are not running with tags that include the cluster size.  For example the `Functional Hardware Medium TCP` stage (https://build.hpdd.intel.com/blue/rest/organizations/jenkins/pipelines/daos-stack/pipelines/daos/branches/PR-8902/runs/13/nodes/65/log/?start=0) is running with:
```
+ ./launch.py -jcrisap -th 1G -ts wolf-138,wolf-139,wolf-140,wolf-141 --nvme=auto:-3DNAND --provider=ofi+tcp pr daily_regression
```
when it should be using:
```
+ ./launch.py -jcrisap -th 1G -ts wolf-138,wolf-139,wolf-140,wolf-141 --nvme=auto:-3DNAND --provider=ofi+tcp pr,hw,medium daily_regression,hw,medium
```

**Review by JohnMalmberg — APPROVED** [review]

**Review by JohnMalmberg — APPROVED** [review]

**Review by daltonbohning — APPROVED** [review]

File: Jenkinsfile [hunk 3]
```diff
@@ -108,7 +168,7 @@ pipeline {
                 expression { !skipStage() }
             }
             parallel {
-                stage('Functional on CentOS 7') {
+                 stage('Functional on CentOS 8') {
```
Comment (brianjmurrell) [inline Jenkinsfile:171]: This really should be:

```suggestion
                 stage('Functional on EL 8') {
```

**Pull request merged**